src/modules/service/CoinMaketService.ts

```diff
@@ -28,34 +31,90 @@ export class CoinMarketService {
-                    logger.info(`CoinMarket: Fetching Job Stated`);
-                    this.coinMarketClient.fetch().then(async (data: IMarketCap) => {
-                        await stoaInstance.putCoinMarketStats(data);
-                        resolve(true);
-                    }).catch((err: any) => {
-                        this.recover(stoaInstance);
-                        reject(err);
-                    });
-                }).catch((err: any) => {
-                    this.recover(stoaInstance);
-                    reject(err);
-                });
+            this.status = false;
+            cron.schedule(`*/${time} * * * * *`, async () => {
+                await this.scheduler(stoaInstance);
+                resolve(true);
             })
+
+        });
+    }
+    public scheduler(stoaInstance: Stoa): Promise<boolean> {
+        return new Promise<boolean>(async (resolve, reject) => {
+            if (this.status === true) {
+                return resolve(true);
+            }
+            let geckoConnection = await this.coinMarketClient.ping();
+            if (!geckoConnection) {
+                logger.warn(`Hints: either api.coingecko.com is available or check you internet connection`);
+                return resolve(true);
+            }
+            else {
+                this.status = true;
+                stoaInstance.ledger_storage.getCoinMarketcap().then(async (rows: any[]) => {
+                    if (!rows[0]) {
+                        await this.recover24hourData(stoaInstance);
+                        this.status = false;
+                        return resolve(true)
+                    }
+                    else {
+                        this.coinMarketClient.fetch().then(async (data: IMarketCap) => {
+                            await stoaInstance.putCoinMarketStats(data);
+                            this.status = false;
+                            return resolve(true);
+                        });
+                    }
+                }).catch(async (err: any) => {
+                    await this.recover24hourData(stoaInstance);
+                    this.status = false;
+                    resolve(true)
+                });
+            }
+        });
+    }
+    /**
+     * @param StoaInstance 
+     * Asynchronously recover 24 hour CoinMarket Data
+     */
+    public recover24hourData(stoaInstance: Stoa): Promise<boolean> {
+        return new Promise<boolean>(async (resolve, reject) => {
+            try {
+                let to = await Time.msToTime(Date.now());
+                let from = await JSBI.subtract(JSBI.BigInt(to.seconds), JSBI.BigInt(60 * 60 * 24));
+                let dt = new Date(to.seconds * 1000);
+                let df = new Date(Number(from.toString()) * 1000)
+                logger.info(`Recovering 24 hour coinmarket cap from: ${df} to: ${dt}`)
+                let marketCap = await this.coinMarketClient.recover(Number(from.toString()), to.seconds);
```

Comment:
err: TypeError: this.coinMarketClient.recover is not a function
Errors are occurring from time to time. (TestCoinGecko)

Reply:
resolved